contrib/vistools/test/test_integration.py

```diff
@@ -0,0 +1,18 @@
+import os
+import pytest
+from filecmp import dircmp
+
+from generate_graphs import generate
+
+CWD = os.path.dirname(os.path.realpath(__file__))
+
+BEAM_COMPARISONS = [(os.path.join(CWD, "resources", "beams.json"), 
+                     os.path.join(CWD, "resources", "output"))]
+
+@pytest.mark.parametrize("beams, expected_output", BEAM_COMPARISONS)
+def test_encoding_end2end(beams, expected_output, tmpdir):
```

Comment:
can you help me understand how this test works? how is tmpdir set and is this actually a temporary directory that gets deleted after this test completes?

Reply:
It's part of pytest's "magic". It's a temp dir fixture described [here](https://docs.pytest.org/en/latest/tmpdir.html). It won't be deleted after the test runs (in case you need to go into it to have a look at the output), but it's created in the system `/tmp` so will eventually be cleaned out.

Reply:
uh cool, didn't know about this.